pkg/cache/partition_info.go

```diff
@@ -477,12 +477,17 @@ func (pi *PartitionInfo) releaseAllocationsForApplication(toRelease *commonevent
 	for _, alloc := range allocationsToRelease {
 		// remove allocation from node
 		node := pi.nodes[alloc.AllocationProto.NodeID]
+		// node could be nil if when release happens, the node is already
+		// removed from the cluster. in this case, we need to make sure
+		// resources are released correctly. otherwise these resources will
+		// be leaked.
```

Comment:
This cannot happen, and is impossible.
When a node is removed the allocations are released by calling `removeNodeAllocations()` at that point they are passed back and should have been released. The queue will get out of sync if we do this twice.